File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline
N = int(input())
dict = {}

for _ in range(N):
    operation = list(input().split())
    order = operation[0]
    
    if order == "add":
        dict[int(operation[1])] = 1
    elif order == "remove":
        if int(operation[1]) in dict:
            del dict[int(operation[1])]
    elif order == "check":
        if int(operation[1]) in dict:
            print(1)
        else:
            print(0)
    elif order == "toggle":
        if int(operation[1]) in dict:
            del dict[int(operation[1])]
        else:
            dict[int(operation[1])] = 1
    elif order == "all":
        dict = {i:1 for i in range(1,21)}
    else:
        dict = {}
        

File: 백준/Silver/11723. 집합/집합.cc
#include <iostream>
#include <vector>
#include <string>

using namespace std;
int n;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    
    cin >> n;
    cin.ignore();
    vector<int> arr(21);

    for (int i = 0; i < n; i++) {
        string order;
        getline(cin, order);
        

        if (order == "all") {
            for (int j = 1; j <= 20; j++) {
                arr[j] = 1;
            }
        } else if (order == "empty") {
            arr = vector<int>(21, 0);
        } else {
            auto it = order.find(" ");
            string cmd = order.substr(0, it);
            int num = stoi(order.substr(it + 1));

            if (cmd == "add") {
                arr[num] = 1;
            } else if (cmd == "remove") {
                arr[num] = 0;
            } else if (cmd == "check") {
                cout << arr[num] << '\n';
            } else if (cmd == "toggle") {
                arr[num] = !arr[num];
            }
        }
    }
    return 0;
}